### PR TITLE
Deprecate support for MPIR

### PR DESCRIPTION
**Pull request by mahrud**
- **deprecated support for MPIR**

### CODE REVIEW

**Comment by d-torrance** [comment]
Why FLINT 2.8.5?  Ubuntu 22.04 still has FLINT 2.8.4, and Macaulay2 still builds fine using FLINT 2.6.3 (which I backported for the  Ubuntu 18.04 and 20.04 PPA's).

**Comment by DanGrayson** [comment]
3  test builds are failing

**Comment by DanGrayson** [comment]
Could you fix the failing test builds?

**Comment by d-torrance** [comment]
Looks good to me, but I suppose we should probably do the same thing for the autotools build.  There's still a bunch of mpir cruft in configure.ac, for example.  I can work on it and push to your branch, if you'd like.

**Comment by mahrud** [comment]
Sounds good!

**Review by d-torrance — APPROVED** [review]
Autotools build should be MPIR-free now too.

File: M2/LICENSING-NOTES [hunk 1]
```diff
@@ -49,4 +48,4 @@ modifications by Paul Zinn-Justin			public domain
 
 for M2 binaries:
 
-We are prevented from using GPL 2 by mpir and mpfr
+We are prevented from using GPL 2 by mpfr
```
Comment (mahrud) [inline M2/LICENSING-NOTES:51]: Huh, [mpfr.org](https://www.mpfr.org/) says:
> MPFR is free. It is distributed under the [GNU Lesser General Public License](https://www.gnu.org/licenses/lgpl-3.0.html) (GNU Lesser GPL), version 3 or later (2.1 or later for MPFR versions until 2.4.x).

I didn't know this file existed.

Comment (d-torrance) [reply M2/LICENSING-NOTES:51]: Apparently GPL-2 and LGPL-3 aren't compatible: https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility

Comment (mahrud) [reply M2/LICENSING-NOTES:51]: Ah, I see.

**Pull request merged**